### ISSUE TITLE
Optimistic shorthand parsing?

### ISSUE THREAD

**Issue by ariarzer** [AndyOGo/stylelint-declaration-strict-value#114]
I'm having trouble using shorthand like `border: 0;`.

I use this config:
```
    'scale-unlimited/declaration-strict-value': [
      [
        '/color/',
        'fill',
        'stroke',
      ], {
        ignoreValues: [
          'currentColor',
          'transparent',
          'inherit',
        ],
      }
    ],
```

If i use `expandShorthand: true` option, I have warning `Expected variable, function or keyword for "0" of "border"`.
But, if i don't use this option, I loss warning, like for `border: 1px solid #eee;`.

Want to make an option for optimistic parsing?

Or I didn’t find it in the documentation, then sorry 😖

Thanks in advance for your reply!)

**Comment by AndyOGo** [comment]
Hi @ariarzer 

That is the expected behavior if you enable `expandShorthand`.
Your provided `/color/` regex will match the expanded `border-color` property and you will get this warning, since `#eee` isn't any of the `ignoredValues`.

Either fix your `border-color` according to your linting rules or be more specific with your `/color/` regex or disable shorthand expansion.

**Comment by AndyOGo** [comment]
Hi @ariarzer 

Sorry, I just re-read your issue.

Regarding `border: 0` gets expanded by `shortcss` to `['border-width: 0', 'border-style: 0', 'border-color: 0']` (without recursion).
You see [`shortcss`](https://github.com/theprotein/shortcss#examples) expansion, does not check if the expanded value is valid CSS.

An explanation why this is like that can be found in that repo too:
https://github.com/AndyOGo/stylelint-declaration-strict-value#css-shorthand-syntax

**Comment by AndyOGo** [comment]
Alright, I found a module which offers to validate CSS property/value pairs, though it is still alpha.
https://github.com/ben-eb/css-values

**Comment by AndyOGo** [comment]
Thank you @ariarzer 

I have created a PR to address you issue with `css-values` module.
It's support for properties is enough for your `border: 0;` case, but may be an issue elsewhere.
All supported properties can be found here:
https://github.com/ben-eb/css-values#property-support

**Comment by AndyOGo** [comment]
Hi @ariarzer 
I just release a patch version [`v1.6.1`](https://github.com/AndyOGo/stylelint-declaration-strict-value/blob/master/CHANGELOG.md#161-2020-10-29).
This should be now working for you.

**Comment by ariarzer** [comment]
Thanks you very much!

**Comment by AndyOGo** [comment]
You are welcome.

Apologize my initial reply, I was distracted and in a hurry and miss-read your message.

Enjoy your day.
